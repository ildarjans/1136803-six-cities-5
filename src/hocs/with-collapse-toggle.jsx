import React from "react";

export const withCollapseToggle = (Component) => {
  class CollapseToggle extends React.PureComponent {
    constructor(props) {
      super(props);
      this.state = {
        collapsed: true
      };

      this.handleCollapseToggle = this.handleCollapseToggle.bind(this);
    }

    handleCollapseToggle() {
      this.setState((state) => ({
        collapsed: !state.collapsed
      }));
    }

    render() {
      const {collapsed} = this.state;
      return <Component
        {...this.props}
        collapsed={collapsed}
        onCollapseToggle={this.handleCollapseToggle}
      />;
    }
  }

  return CollapseToggle;
};
