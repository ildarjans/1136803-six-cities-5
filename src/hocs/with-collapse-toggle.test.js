import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import {withCollapseToggle} from "./with-collapse-toggle";

const mockedComponent = ({collapsed, onCollapseToggle}) => {
  return (
    <button
      id="btn"
      type="button"
      onClick={onCollapseToggle}
    >
      {`Initial collapsed state is ${collapsed}`}
    </button>
  );
};

mockedComponent.propTypes = {
  collapsed: PropTypes.bool.isRequired,
  onCollapseToggle: PropTypes.func.isRequired
};

it(`Should CollapseToggle HOC render correctly`, () => {
  const WithLoadingComponent = withCollapseToggle(mockedComponent);
  const tree = renderer.create(
      <WithLoadingComponent/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
