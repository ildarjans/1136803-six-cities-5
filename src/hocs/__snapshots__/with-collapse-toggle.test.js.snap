// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CollapseToggle HOC render correctly 1`] = `
<button
  id="btn"
  onClick={[Function]}
  type="button"
>
  Initial collapsed state is true
</button>
`;
